workers/python/kitchen_sink.py

```diff
@@ -81,38 +99,64 @@ def workflow_query(arg: str) -> str:
 
 @workflow.defn(name="kitchenSink")
 class KitchenSinkWorkflow:
-    def __init__(self) -> None:
-        self._signal_name: str = ""
-        self._pending_actions: asyncio.Queue[Action] = asyncio.Queue()
-        self._signal_recieved: Dict[str, asyncio.Queue[Any]] = {}
-
-    @workflow.signal(dynamic=True)
-    async def handle_signal(self, signal_name: str, *varargs) -> None:
-        if signal_name == self._signal_name:
-            await self._pending_actions.put(varargs[0])
-        else:
-            await self._signal_recieved.setdefault(signal_name, asyncio.Queue()).put(
-                varargs[0]
-            )
-
     @workflow.run
-    async def run(self, input: WorkflowParams) -> Any:
-        self._signal_name = input.action_signal
+    async def run(self, input: Optional[WorkflowParams]) -> Any:
         workflow.logger.info("Started kitchen sink workflow {}".format(input))
-        for action in input.actions:
-            should_return, return_value = await self.handle_action(input, action)
+        if not input:
+            return None
+        if input.action_set and input.action_set.actions:
+            should_return, return_value = await self.handle_action_set(
+                input, input.action_set
+            )
             if should_return:
                 return return_value
 
-        if input.action_signal != "":
-            while not self._pending_actions.empty():
-                action = await self._pending_actions.get()
-                should_return, return_value = await self.handle_action(input, action)
+        if input.action_set_signal:
+            action_set_queue: asyncio.Queue[ActionSet] = asyncio.Queue()
+
+            def enqueue(item: ActionSet) -> None:
+                action_set_queue.put_nowait(item)
+
+            workflow.set_signal_handler(input.action_set_signal, enqueue)
+            while True:
+                action_set = await action_set_queue.get()
+                should_return, return_value = await self.handle_action_set(
+                    input, action_set
+                )
                 if should_return:
                     return return_value
-                self._pending_actions.task_done()
 
-        return None
+        return return_value
+
+    async def handle_action_set(
+        self, input: WorkflowParams, action_set: ActionSet
+    ) -> Tuple[bool, Any]:
+        should_return = False
+        return_value = None
+        # If these are non-concurrent, just execute and return if requested
+        if not action_set.concurrent:
+            for action in action_set.actions:
+                should_return, return_value = await self.handle_action(input, action)
```

Comment:
Doesn't seem like there that much value in having the actions return anything.
Thinking of just dropping this functionality.
Also would be nice to be able to handle a stream of signals instead of batching but I think this is good enough for now.

Reply:
Return value is fine, but we can remove later if we need. We do handle a stream of signals unless I misunderstand.

Reply:
We block until the action or set is complete before processing the next signal, streaming would allow receiving more work while processing is in progress.

Reply:
Ah, I meant the `action_set_signal`. But yes, when we need that functionality we can add it.

We need to be very careful guessing at things we might need. This needs to be an evolving system, not one based on guesses. The only reason this has the features it does now is because I needed them for integration testing the Python SDK, no more.